testplan/web_ui/testing/src/AssertionPane/AssertionTypes/basicAssertionUtils.js

```diff
@@ -224,176 +586,84 @@ function prepareBasicContent(assertion) {
     rightContent: assertion.first,
     postTitle: null,
     postContent: null,
-  };
+  }
 
-  var uid = hashCode(JSON.stringify(assertion));
+  // Fan out to the relevant function to prepare content for each assertion
+  // type.
+  switch (assertion.type) {
```

Comment:
Yay! its much nicer

Reply:
Thanks! Using switch to "fan out" an incoming object into subtypes is quite a common pattern, I've seen it used before in the context of receiving a message and switching based on the message type to decide how to handle it. I think it's a lot more readable than if/else (sadly Python has no switch statement - you can simulate them using a dict of functions though).

I remember reading about a design principle a while ago that you should generally split functions into three categories - 1) "deciding" functions that branch based on some condition, 2) "Pure" functions that just calculate a value based on some input with no side effects, and 3) "Business logic" functions that perform some well-defined side effect (e.g. writing to a file). Here I've tried to only use 1) (for the main "switch" function) and 2) (for the prepareContent helpers).